aten/src/ATen/native/Convolution.cpp

```diff
@@ -1638,7 +1638,9 @@ std::tuple<Tensor, Tensor, Tensor> convolution_backward(
       std::array<bool, 2> input_weight_output_mask = {output_mask[0], output_mask[1]};
       std::tie(backend_grad_input, backend_grad_weight) = cudnn_convolution_backward_stub(
           input.device().type(),
-          input.contiguous(backend_memory_format), grad_output, weight, params.padding, params.stride,
+          // Only make input contiguous when it is necessary for the backwards computation
+          output_mask[1] ? input.contiguous(backend_memory_format) : input,
+          grad_output, weight, params.padding, params.stride,
```

Comment:
My understanding is that this covers the case where:
* `weight` is `channels_last`
* `input` is **not** `channels_last`
* `grad_output` is **not** `channels_last`
* `grad_weight` is being computed, which involves input & grad_output

We want channels_last to be used, but `cudnn_convolution_backward_weight` bases its decision only on input / grad_output:
https://github.com/pytorch/pytorch/blob/f45e217c0158c247772f6491a18662cc972c29cd/aten/src/ATen/native/cudnn/ConvShared.cpp#L387

Explicitly setting `input.contiguous(backend_memory_format)` before the call to `cudnn_convolution_backward_weight` will trigger the above logic to return channels_last.

Interestingly, we *don't* need analogous logic for `cudnn_convolution_backward_input` to do the right thing because of:
https://github.com/pytorch/pytorch/blob/f45e217c0158c247772f6491a18662cc972c29cd/aten/src/ATen/native/cudnn/ConvShared.cpp#L430

and

https://github.com/pytorch/pytorch/blob/f45e217c0158c247772f6491a18662cc972c29cd/aten/src/ATen/native/cudnn/ConvShared.cpp#L470

ensuring channels_last is correctly used in this case:
* `weight` is **not** `channels_last`
* `input` is `channels_last`
* `grad_output` is **not** `channels_last` (will be made channels_last due to `input.suggest_memory_format()` calls above)
* `grad_input` is being computed, which involves weight & grad_output

TL;DR: Changes lgtm :)